### ISSUE TITLE
Add resource for Magic Firewall API (using Rulesets API)

### ISSUE THREAD

**Issue by cehrig** [cloudflare/terraform-provider-cloudflare#880]
### Description

We should add a resource for managing Magic Transit Firewall Rulesets aka. Magic Firewall

### Use-cases

- Create, Delete, Update Magic Firewall Rules

### Potential Terraform configuration

```hcl
resource "cloudflare_firewall_ruleset" "example" {
  kind = "root"
  phase = "magic_transit"
  name = "static_mitigations"
  description = "Static Mitigations"

  rule {
    action = "allow"
    expression = "udp.dstport in { 32768..65535 }"
    description = "Allow UDP Ephemeral Ports"
    enabled = true
  }

  rule {
    action = "block"
    expression = "ip.len >= 0"
    description = "Deny All"
    enabled = true
  }
}
```

### References

cloudflare-go interfaces are going to be implemented via https://github.com/cloudflare/cloudflare-go/pull/558
I'm going to pick this up as soon as I'm finished with the cloudflare-go changes

**Comment by jacobbednarz** [comment]
Nice! Thanks for the heads up @cehrig! This would be a much welcome addition 😄

**Comment by jacobbednarz** [comment]
closed by ##884
